### DISCUSSION TITLE
Force upgrades

### DISCUSSION THREAD

**Discussion by mmurali88** [MobileNativeFoundation/discussions#187] [General]
I'm curious to hear from folks about what their force upgrade strategies are. I have seen multiple apps do mandatory force upgrades while I have seen other apps like Target where I have never seen not even a single one.

**Comment by philipphofmann** [comment]
At mySugr, where I worked some time ago, we had a manually built dialog blocking the user from doing anything in the app if they didn't update to a specific version. The dialog was configurable with a simple JSON stored on the backend, which the app checked on every launch or so. With that approach, we could force users to update to a specific app version by changing the JSON on the backend. In three years of working there, I think we only used it once or twice. We implemented this dialog cause mySugr is an app with medical components, and a critical bug could harm the user's health in the worst case.

**Comment by bill350** [comment]
At Back Market, we use CloudBees Feature Management and a dedicated feature flag to enable/disable the force update.

It helps us to target specific iOS or/and Android versions, create advanced conditions of force update (by region or other user properties), and not necessarily make it enabled for 100% of users.

As @philipphofmann said, it also displays a full-screen modal that you cannot dismiss.

It is vital for security purposes/big API changes or depreciations (avoid clients connecting to old APIs, for instance), or for breaking changes where we have no choice but to ask users to update.
